plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java

```diff
@@ -68,16 +78,18 @@ public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
             LocalDateTime timestamp;
             if (value instanceof String) {
                 String valueString = (String) value;
-                Long epochMillis = Longs.tryParse(valueString);
-                if (epochMillis != null) {
-                    timestamp = LocalDateTime.ofInstant(Instant.ofEpochMilli(epochMillis), UTC);
+                try {
+                    timestamp = LocalDateTime.from(formatter.parse(valueString));
                 }
-                else {
+                catch (IllegalArgumentException e) {
+                    // Compatible for Elasticsearch6
+                    // Docvalue_fields query for Elasticsearch6 will always return ISO_DATE_TIME format
```

Comment:
Is it possible to parse using ISO_DATE_TIME in case of Elasticsearch 6 without catching exceptions?

Reply:
I can't find any other methods to help us determine if it's in the case of ES6.  I think we can only check if the actual date value matches ISO_DATE_TIME, before or after using formatter, both needs try-catch phrases. Do you have any ideas?